### ISSUE TITLE
`mp_importantColor` crashes on some images.

### ISSUE THREAD

**Issue by tettoffensive** [mixpanel/mixpanel-iphone#450]
@malectro I was looking at using your algorithm on some images it crashes in the loop because the data array is accessed beyond the bounds. I can't really figure out why as the various CGBitmapContextGet* functions return the same values.

```
   // only attempt to quantize the header
    NSUInteger l = (NSUInteger) ceil(self.size.width * 124.0f);
    for (NSUInteger i = 40 * 640; i < l; i++) {
        uint8_t red = data[i * 4 + 2];
        uint8_t green = data[i * 4 + 1]; // **** CRASHES HERE WHEN i = 40*640 ***
        uint8_t blue = data[i * 4 + 0];
        NSInteger hexColor = (red >> 2) + ((green >> 2) << 6) + ((blue >> 2) << 12);

        if (hexColor > 0 && hexColor < 2621443 && red + green + blue < 255 + 255 + 200 && red != blue && blue != green && green != red) {
            colorIndices[hexColor]++;
        }
    }
```

**Comment by samgreen** [comment]
Thanks for reporting this. Do you have a specific image that crashes so we can start testing a fix?

**Comment by tettoffensive** [comment]
I don't know exactly what specific image it is happening to since it's not happening on my device, but others have it. But it happens when getting cropping an image like this:
```
CGImageRef imageRef = CGImageCreateWithImageInRect([colorReferenceImage CGImage], CGRectMake(0, colorReferenceImage.size.height-20, colorReferenceImage.size.width, 20)); // bottom 20px used for dominant color
UIImage *bottomOfImage = [UIImage imageWithCGImage:imageRef];
CGImageRelease(imageRef);
UIColor *dominantColor = [bottomOfImage mp_importantColor];
```

I think there must be something about the image that makes the hardcoded numerical values go beyond the bounds of the data array.

**Comment by samgreen** [comment]
Yeah we have re-written the implementation to not use any hardcoded numbers and call out to the `CGBitmap` helper functions. Was just hoping for a test case. I'll update this issue when the PR is ready, expect the fix in the next release.

Thanks!

**Comment by samgreen** [comment]
This is resolved in v2.9.5 which is now released.
